packages/@aws-cdk/aws-autoscaling/lib/lifecycle-hook-target.ts

```diff
@@ -1,26 +1,61 @@
-
-import { ILifecycleHook } from './lifecycle-hook';
+// eslint-disable-next-line import/order
+import { LifecycleHook } from './lifecycle-hook';
+import * as iam from '@aws-cdk/aws-iam';
 
 // keep this import separate from other imports to reduce chance for merge conflicts with v2-main
 // eslint-disable-next-line no-duplicate-imports, import/order
-import { Construct } from '@aws-cdk/core';
+import * as constructs from 'constructs';
 
 /**
- * Interface for autoscaling lifecycle hook targets
+ * Options needed to bind a target to a lifecycle hook.
+ * [disable-awslint:ref-via-interface] The lifecycle hook to attach to and an IRole to use
  */
-export interface ILifecycleHookTarget {
+export interface BindHookTargetOptions {
   /**
-   * Called when this object is used as the target of a lifecycle hook
+   * The lifecycle hook to attach to.
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly lifecycleHook: LifecycleHook;
+  /**
+   * The role to use when attaching to the lifecycle hook.
+   * [disable-awslint:ref-via-interface]
+   * @default: a role is not created unless the target arn is specified
    */
-  bind(scope: Construct, lifecycleHook: ILifecycleHook): LifecycleHookTargetConfig;
+  readonly role?: iam.IRole;
 }
 
 /**
- * Properties to add the target to a lifecycle hook
+ * Result of binding a lifecycle hook to a target.
  */
 export interface LifecycleHookTargetConfig {
   /**
-   * The ARN to use as the notification target
+   * The IRole that was used to bind the lifecycle hook to the target
+   */
+  readonly createdRole: iam.IRole;
+  /**
+   * The targetArn that the lifecycle hook was bound to
    */
   readonly notificationTargetArn: string;
 }
+
+/**
+ * Interface for autoscaling lifecycle hook targets
+ */
+export interface ILifecycleHookTarget {
+  /**
+   * Called when this object is used as the target of a lifecycle hook
+   * @param options [disable-awslint:ref-via-interface] The lifecycle hook to attach to and a role to use
+   */
+  bind(scope: constructs.Construct, options: BindHookTargetOptions): LifecycleHookTargetConfig;
+}
+
+export function createRole(scope: constructs.Construct, _role?: iam.IRole) {
```

Comment:
This is a hacky private API (accidentally public to TS users), that I'm not sure this is worth sharing between `aws-autoscaling` and `aws-autoscaling-hooktargets` in this way. Just move it to the other library I would say.